yb-voyager/src/types/query_stats.go

```diff
@@ -27,3 +33,64 @@ type QueryStats struct {
 	MinExecTime     float64 `json:"min_exec_time" db:"min_exec_time"`
 	MaxExecTime     float64 `json:"max_exec_time" db:"max_exec_time"`
 }
+
+// MergeQueryStatsBasedOnQuery merges query stats with the same query text.
+// Used to aggregate stats from multiple source nodes (primary + replicas).
+// Mirrors the logic of pgss.MergePgStatStatementsBasedOnQuery but operates on QueryStats directly
```

Comment:
I don't fully understand why can't we reuse MergePgStatStatementsBasedOnQuery. Let's discuss this